streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyTestRunResource.java

```diff
@@ -98,6 +99,25 @@ public Response testRunTopology (@Context UriInfo urlInfo,
         throw EntityNotFoundException.byId(topologyId.toString());
     }
 
+    @POST
+    @Path("/topologies/{topologyId}/actions/killtest/{historyId}")
+    @Timed
+    public Response killTestRunTopology(@Context UriInfo urlInfo,
+                                        @PathParam("topologyId") Long topologyId,
+                                        @PathParam("historyId") Long historyId) throws Exception {
+        Topology topology = catalogService.getTopology(topologyId);
+        TopologyTestRunHistory history = catalogService.getTopologyTestRunHistory(historyId);
+
+        if (topology == null) {
+            throw EntityNotFoundException.byName("Topology with ID " + topologyId.toString());
+        } else if (history == null) {
+            throw EntityNotFoundException.byName("TopologyTestRunHistory with ID " + historyId.toString());
+        }
+
+        boolean flagged = actionsService.killTestRunTopology(topology, history);
+        return WSUtils.respondEntity(Collections.singletonMap("flagged", flagged), OK);
```

Comment:
Since this just marks for kill, when does UI consider kill as successful ? It needs to query the status?

Reply:
Yes this API just triggers force kill, and loop for polling test history and events should be continue until history provides 'finish' to true.